pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BookkeeperBucketSnapshotStorage.java

```diff
@@ -172,12 +157,25 @@ private CompletableFuture<LedgerHandle> createLedger(String bucketKey, String to
                     if (rc != BKException.Code.OK) {
                         future.completeExceptionally(bkException("Create ledger", rc, -1));
                     } else {
+                        ledgerHandleCache.put(handle.getId(), handle);
```

Comment:
You shouldn't close it if you want to cache a writable ledger handle.

Reply:
I don't suggest caching this handle here. since you want the bucket(ledger) to be immutable, you should close it.

Reply:
It makes sense to don't caching this handle here.